### ISSUE TITLE
Error Prefix Not Shown With Error Messages

### ISSUE THREAD

**Issue by danielhelfand** [knative/client#885]
### Bug report

An error prefix is not displayed with error messages. Only the error message is shown.

Cobra handles error messages by default if `SilenceErrors` is false, but `kn` handles errors in main.go and [sets `SilenceErrors` to true](https://github.com/knative/client/blob/ff350fb6a70f4903e8744654fa2e470ea048811d/pkg/kn/core/root.go#L154) for the root command. 

By removing the `SilenceErrors` from the root command, there would be no need to print error messages in main.go, and it could be rewrote as follows:

```go
func main() {
	defer cleanup()
	rand.Seed(time.Now().UnixNano())
	
        kn, err := core.NewDefaultKnCommand()
        if err != nil {
            os.Exit(1)
        }

	if err = kn.Execute(); err != nil {
	    os.Exit(1)
	}
}
```

Then Cobra would properly attach the error prefix to messages by default for all errors. 

### Expected behavior

An error prefix should be displayed before error messages. 

```
kn service l
Error: unknown sub-command "l" for "kn service"
```

### Steps to reproduce the problem

Produce an error of any kind:

```
kn service l
unknown sub-command "l" for "kn service"
```

### kn version

What's currently in master as of v0.15.1.

**Comment by navidshaikh** [comment]
Thanks @danielhelfand! Linking it with generic tracker for error handling issue https://github.com/knative/client/issues/213 to be addressed.

**Comment by rhuss** [comment]
I think this has been fixed already in the latest PR merged with #877 and improved in #887 . There is now only a single point where the error is printed (in `main.go`). It's used to customize the error message, I found the printing out of errors at different levels in the stack confusing (which sometimes lead to double error printing when e.g. the error message was printed but still the error was propagated upwards). `SilenceError` in combination with a  single point of error printing is imo the better solution.

@danielhelfand could you please check, whether the #887 fixes the issues that you have encountered ?
